File: app/src/main/java/com/imangazalievm/bubbble/data/shots/ImagesRepository.java
package com.imangazalievm.bubbble.data.shots;

import com.imangazalievm.bubbble.data.global.filesystem.UrlImageSaver;

import javax.inject.Inject;

import io.reactivex.Completable;

public class ImagesRepository {

    private UrlImageSaver urlImageSaver;

    @Inject
    public ImagesRepository(UrlImageSaver urlImageSaver) {
        this.urlImageSaver = urlImageSaver;
    }

    public Completable saveImage(String shotImageUrl) {
        return Completable.create(e -> {
            urlImageSaver.saveImage(shotImageUrl);
            e.onComplete();
        });
    }

}

File: app/src/main/java/com/imangazalievm/bubbble/data/shots/ImagesRepository.kt
package com.imangazalievm.bubbble.data.shots

import com.imangazalievm.bubbble.data.global.filesystem.UrlImageSaver
import io.reactivex.Completable
import io.reactivex.CompletableOnSubscribe
import io.reactivex.CompletableEmitter
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import javax.inject.Inject

class ImagesRepository @Inject constructor(
    private val urlImageSaver: UrlImageSaver
) {

    suspend fun saveImage(shotImageUrl: String) = withContext(Dispatchers.IO) {
        urlImageSaver.saveImage(shotImageUrl)
    }

}